File: tests/transpiler/x/fs/user_type_literal.fs
// Generated 2025-07-21 18:37 +0700

type Person = {
    mutable name: string
    mutable age: int
}
type Book = {
    mutable title: string
    mutable author: Person
}
let book: Book = { title = "Go"; author = { name = "Bob"; age = 42 } }
printfn "%s" (string (book.author.name))

File: tests/transpiler/x/pl/user_type_literal.pl
:- initialization(main).
:- style_check(-singleton).

main :-
    Book = _{tag: "Book", title: "Go", author: _{tag: "Person", name: "Bob", age: 42}},
    get_dict(name, get_dict(author, Book, R), V1), writeln(V1).
